autopilot/stim/sound/jackclient.py

```diff
@@ -145,51 +144,120 @@ def __init__(self, name='jack_client'):
         globals()['CONTINUOUS_QUEUE'] = self.continuous_q
         globals()['CONTINUOUS_LOOP'] = self.continuous_loop
 
-        self.logger = None
-
-
+        self.logger = init_logger(self)
+        
+        # Something calls process() before boot_server(), so this has to
+        # be initialized
+        self.mono_output = True
 
     def boot_server(self):
         """
         Called by :meth:`.JackClient.run` to boot the server upon starting the process.
 
-        Activates the client and connects it to the number of outports
-        determined by `prefs.get('NCHANNELS')`
+        Activates the client and connects it to the physical speaker outputs
+        as determined by `prefs.get('OUTCHANNELS')`.
+
+        This is the interpretation of OUTCHANNELS:
+        * empty string
+            'mono' audio: the same sound is always played to all channels. 
+            Connect a single virtual outport to every physical channel.
+            If multi-channel sound is provided, raise an error.
+        * a single int (example: J)
+            This is equivalent to [J].
+            The first virtual outport will be connected to physical channel J.
+            Note this is NOT the same as 'mono', because only one speaker
+            plays, instead of all speakers.
+        * a list (example: [I, J])
+            The first virtual outport will be connected to physical channel I.
+            The second virtual outport will be connected to physical channel J.
+            And so on.    
+            If 1-dimensional sound is provided, play the same to all speakers
+            (like mono mode).
+            If multi-channel sound is provided and the number of channels
+            is different form the length of this list, raise an error.        
 
         :class:`jack.Client` s can't be kept alive, so this must be called just before
         processing sample starts.
         """
-
+        ## Parse OUTCHANNELS into listified_outchannels and set `self.mono_output`
+        # Get the pref
+        outchannels = prefs.get('OUTCHANNELS')
+        
+        # This generates `listified_outchannels`, which is always a list
+        # It also sets `self.mono_output` if outchannels is None
+        if outchannels == '':
+            # Mono mode
+            listified_outchannels = []
+            self.mono_output = True
+        elif not isinstance(outchannels, list):
+            # Must be a single integer-like thing
+            listified_outchannels = [int(outchannels)]
+            self.mono_output = False
+        else:
+            # Already a list
+            listified_outchannels = outchannels
+            self.mono_output = False
+        
+        ## Initalize self.client
+        # Initalize a new Client and store some its properties
+        # I believe this is how downstream code knows the sample rate
```

Comment:
a client is connected in the `__init__` method that connects to determine the fs and then dies, another client is booted in `boot_server` because the clients dont survive the process fork